listenbrainz_spark/path.py

```diff
@@ -5,39 +5,47 @@
+CHECKPOINT_DIR = os.path.join('/', 'checkpoint')
 
+################################################
+# Recommendation `recording` absolute path/dir #
+################################################
 
-# Parent directory containing data used in and generated by recommendation engine.
-RECOMMENDATION_PARENT_DIR = os.path.join('/', 'recommendation')
-# Directory containing dataframes to be used in generating recommendations (not necessarily).
-DATAFRAME_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'dataframe')
-# Directory containing model metadata and the real models to be used for generating recommendations.
-MODEL_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'model')
+# Parent directory containing data used in and generated by `recording` recommendation engine.
+RECOMMENDATION_RECORDING_PARENT_DIR = os.path.join('/', 'recommendation')
+# Directory containing dataframes to be used in generating `recording` recommendations (not necessarily).
+RECOMMENDATION_RECORDING_DATAFRAME_DIR = os.path.join(RECOMMENDATION_RECORDING_PARENT_DIR, 'dataframe')
+# Directory containing model metadata and the real models to be used for generating `recording` recommendations.
+RECOMMENDATION_RECORDING_MODEL_DIR = os.path.join(RECOMMENDATION_RECORDING_PARENT_DIR, 'model')
 # Directory containing candidate sets to be used for generating recommendations.
-CANDIDATE_SET_DIR = os.path.join(RECOMMENDATION_PARENT_DIR, 'candidate_set')
-# Directory containing RDD checkpoints to break lineage while using iterative algorithms.
-CHECKPOINT_DIR = os.path.join('/', 'checkpoint')
-# Directory to save best models
-DATA_DIR = os.path.join(MODEL_DIR, 'data')
+RECOMMENDATION_RECORDING_CANDIDATE_SET_DIR = os.path.join(RECOMMENDATION_RECORDING_PARENT_DIR, 'candidate_set')
+# Directory to save best models for `recording` recommendations.
+RECOMMENDATION_RECORDING_DATA_DIR = os.path.join(RECOMMENDATION_RECORDING_MODEL_DIR, 'data')
 
-# Absolute path to dataframes used in processing raw data/listens.
-USERS_DATAFRAME_PATH = DATAFRAME_DIR + '/' + 'users_df.parquet'
-RECORDINGS_DATAFRAME_PATH = DATAFRAME_DIR + '/' + 'recordings_df.parquet'
-# Absolute path to processed data/listens ready to be trained.
-PLAYCOUNTS_DATAFRAME_PATH = DATAFRAME_DIR + '/' + 'playcounts_df.parquet'
-# Absolute path to similar artist relation.
-SIMILAR_ARTIST_DATAFRAME_PATH = SIMILAR_ARTIST_DIR + '/' + 'artist_credit_artist_credit_relations.parquet'
-# Absolute path to candidate sets.
-TOP_ARTIST_CANDIDATE_SET = os.path.join(CANDIDATE_SET_DIR, 'top_artist', 'top_artist.parquet')
-SIMILAR_ARTIST_CANDIDATE_SET = os.path.join(CANDIDATE_SET_DIR, 'similar_artist', 'similar_artist.parquet')
+# Absolute path to dataframes used in processing raw data/listens for `recording` recommendations.
+RECOMMENDATION_RECORDING_USERS_DATAFRAME = RECOMMENDATION_RECORDING_DATAFRAME_DIR + '/' + 'users_df.parquet'
+RECOMMENDATION_RECORDING_RECORDINGS_DATAFRAME = RECOMMENDATION_RECORDING_DATAFRAME_DIR + '/' + 'recordings_df.parquet'
```

Comment:
@mayhem I have prepended `RECOMMENDATIONS_RECORDING` to path names. Every other path name makes sense to me except his one, the `recordings_dataframe` already has `recordings` in it, if I prepend `RECOMMENDATIONS_RECORDING` to it, the result is `RECOMMENDATION_RECORDING_RECORDINGS_DATAFRAME` which seems a bit off.  I haven't updated the path names in recommendation scripts, let's sort this out first?

Reply:
Why not just go with RECOMMENDATION_RECORDING_DATAFRAME ?